webapp/app.py

```diff
@@ -64,6 +65,14 @@
 discourse_docs.init_app(app)
 
 
+@app.route("/docs/_cache_clear")
```

Comment:
I don't like the idea of it being public. The other issue is that we have 5 replicas on production, and unfortunately each of them has an independent cache.

Reply:
Me either, but don't have a good way of protecting it and I really want to remove the stale content ASAP.

The multiple replicas already cause issues with different caches and a changing upstream Discourse (since they'll have different points on the timeline)

Reply:
If you are in a hurry the best you can do right now is to go to Jenkins a redeploy the latest version.

The HTTP method for something like this should be a "POST" or "DELETE" not "GET" in my opinion, and the replicas are a real problem because how do you plan to execute it on all of them?

Reply:
> If you are in a hurry the best you can do right now is to go to Jenkins a redeploy the latest version.
> 

That didn't work. Are you sure that when deploying a docker image with the same label/tag as is already deployed causes a redeploy? In our experience it didn't - compare and contrast the jenkins jobs 159 (new build) and 160 (redeploy) the latter doesn't show any "deployment "maas-io" successfully rolled out"

> The HTTP method for something like this should be a "POST" or "DELETE" not "GET" in my opinion, and the replicas are a real problem because how do you plan to execute it on all of them?

A loop in bash

> 
> I like the idea of having an endpoint to clean it don't get me wrong but seems we are doing it without considering all of it. We could start by reducing the amount of time we cache, maybe even asking IS for a redis instance or other type of db so all replicas use the same db.

Don't let perfect be the enemy of good enough.

The 24h never-invalidated cache is causing real problems for anybody that is editing the docs, it makes it impossible to verify/QA any work you've done and exposes users to horrible inconsistencies (since you could have index document cached at version N, whilst you make version N + 1, N + 2, to align with document D which didn't exist in version N).


I challenge anybody to spend more than a few days working in a system that arbitrarily has parts of the data store varying between 0 and 24h old and see how it feels (it's horrible).

This was sold to us as a workaround, if that's true then we should be able to add another workaround on top (like this cache clearing).